apps/docker/dev-server.Dockerfile

```diff
@@ -0,0 +1,146 @@
+#syntax=docker/dockerfile:1.2
+
+#================================================================
+#
+# DEV: Stage used for the development environment
+#      and the locally built services
+#
+#=================================================================
+
+FROM ubuntu:20.04 as base
+ARG USER_ID=1000
+ARG GROUP_ID=1000
+ARG DOCKER_USER=acore
+
+LABEL description="AC base image for dev containers"
+
+# List of timezones: http://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+
+ENV DOCKER=1
+
+# set timezone environment variable
+ENV TZ=Etc/UTC
+
+# set noninteractive mode so tzdata doesn't ask to set timezone on install
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Do not use acore dashboard to install
+# since it's not cacheable by docker
+RUN apt-get update && apt-get install -y gdb gdbserver git dos2unix lsb-core sudo curl unzip \
+  make cmake clang libmysqlclient-dev \
+  libboost-system1.7*-dev libboost-filesystem1.7*-dev libboost-program-options1.7*-dev libboost-iostreams1.7*-dev \
+  build-essential libtool cmake-data openssl libgoogle-perftools-dev google-perftools \
+  libssl-dev libmysql++-dev libreadline6-dev zlib1g-dev libbz2-dev mysql-client \
+  libncurses5-dev ccache \
+  && rm -rf /var/lib/apt/lists/*
```

Comment:
Wouldn't be smarter to have a single docker file to share the part of setting at least the OS configuration for the building part?

Since this is essentially the same between the dev server and the build process for the production images.

I'm always trying to avoid having lot of stuff to maintain, especially if we can't use the acore.sh here that would have been our single-source-of-truth for the dependency installation

Reply:
With my latest commits, I've removed the dev-server, as I have a hard time figuring out where it fits in.

On my machine rebuilds are always less than a minute with the way the docker build stages are set up. If code changes, docker ONLY re-compiles the programs and adds it into the proper container

1. skeleton
  a. build
  b. runtime
    i. authserver # These 3 get binaries from the build step
    ii. worldserver
    iii. db-import
  c. client-data

Reply:
dev server is used for the vscode devcontainer setup, where you can work within a docker container directly

Reply:
I wouldn't even have a separated db if possible, there's not really a reason for that I guess